src/meta/src/stream/refresh_manager.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::anyhow;
+use risingwave_common::catalog::{DatabaseId, TableId};
+use risingwave_pb::meta::{RefreshRequest, RefreshResponse};
+use thiserror_ext::AsReport;
+
+use crate::barrier::{BarrierScheduler, Command};
+use crate::manager::MetadataManager;
+use crate::{MetaError, MetaResult};
+
+/// Manager responsible for handling refresh operations on refreshable tables
+pub struct RefreshManager {
+    metadata_manager: MetadataManager,
+    barrier_scheduler: BarrierScheduler,
+}
+
+impl RefreshManager {
+    /// Create a new `RefreshManager` instance
+    pub fn new(metadata_manager: MetadataManager, barrier_scheduler: BarrierScheduler) -> Self {
+        Self {
+            metadata_manager,
+            barrier_scheduler,
+        }
+    }
+
+    /// Execute a refresh operation for the specified table
+    ///
+    /// This method:
+    /// 1. Validates that the table exists and is refreshable
+    /// 2. Sends a refresh command through the barrier system
+    /// 3. Returns the result of the refresh operation
+    pub async fn refresh_table(&self, request: RefreshRequest) -> MetaResult<RefreshResponse> {
```

Comment:
When manually triggering a refresh command, do we need to send the loadFinish command as well?

Reply:
The `loadFinish` command is triggered after SourceExec reporting data load finished. added in #22620

Reply:
I see.